### PR TITLE
Add Pester tests to cover functionality added in Chocolatey CLI 2.3.0

### PR DESCRIPTION
**Pull request by gep13**
## Description Of Changes

A number of new Pester tests have been added to cover functionality that has been added.

## Motivation and Context

A lot of new functionality has been added into the develop branch ahead of the 2.3.0 CLI release.

We need to add tests to verify all of this new functionality, and to ensure that there are no regressions going forward.

This also includes a minor bugfix for an issue that surfaced in our test environment.

## Testing

These tests have been validated locally, but we will also kick off some tests runs once this PR is raised.

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [x] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue

N/A

### CODE REVIEW

File: tests/pester-tests/commands/choco-source.Tests.ps1
Comment (corbob) [inline tests/pester-tests/commands/choco-source.Tests.ps1:75]: The comment seems counter to what the action actually is. Although it's not necessary to assign `Invoke-Choco` to `$null`, it's become the norm in the tests, so let's keep it going here.

```suggestion
            # Ensure source is already added
            $null = Invoke-Choco $CurrentCommand add --name "already-exists" --source "https://somewhere/out/there/"

```

File: tests/pester-tests/features/PythonSource.Tests.ps1
Comment (corbob) [inline tests/pester-tests/features/PythonSource.Tests.ps1:43]: If I recall from our local vagrant testing, pip wasn't being found on the path there, but these tests didn't need it to be because pip is never attempted to be run (I think python3 is still needed to be installed because Chocolatey CLI will require the package, but I seem to recall the path doesn't matter here).
```suggestion
```

File: tests/helpers/common-helpers.psm1
Comment (corbob) [inline tests/helpers/common-helpers.psm1:11]: @gep13 this isn't a particularly elegant solution to the issue we were seeing on Test Kitchen, but it does solve the issue seen there.

**Review by corbob — APPROVED** [review]
@gep13 there is nothing here that absolutely must be changed, so I'm going to add an approval on here. I'll hold off on merging it until we can have a discussion on Monday, or you can merge it before that if desired.

The 2 suggested changes I included, are not blockers, the tests should work just fine whether it's changed or not. And the third comment is specifically to draw your attention to a change I added so that Test Kitchen can run the tests successfully.

**Comment by corbob** [comment]
@gep13 I've switched this over to draft, and force pushed a rebase with an additional troubleshooting commit. draft is so we don't merge it without removing the troubleshooting commit.

I'll link you the Team City build in Slack when it starts.

File: tests/helpers/common-helpers.psm1
Comment (gep13) [reply tests/helpers/common-helpers.psm1:11]: @corbob @vexx32 we need to back out this change, and find the solution to the underlying problem.

Comment (corbob) [reply tests/helpers/common-helpers.psm1]: I've backed this out in a revert commit. We can rebase it away before merge.

File: tests/pester-tests/commands/choco-source.Tests.ps1
Comment (gep13) [reply tests/pester-tests/commands/choco-source.Tests.ps1]: As discussed, I have fixed this up.

File: tests/pester-tests/commands/choco-source.Tests.ps1
Comment (corbob) [inline tests/pester-tests/commands/choco-source.Tests.ps1:568]: It's not likely to be there already, but just in case somehow it's there, let's make sure it's enabled (and this makes it the same as the already-disabled test).

```suggestion
            Invoke-Choco $CurrentCommand add --name "already-enabled" --source "https://somewhere/out/there/"
            Invoke-Choco $CurrentCommand enable --name "already-enabled"
```

**Review by corbob — CHANGES_REQUESTED** [review]
Just a couple of changes, and then I think we're good to merge.

**Review by corbob — CHANGES_REQUESTED** [review]

File: tests/pester-tests/commands/choco-source.Tests.ps1
Comment (corbob) [reply tests/pester-tests/commands/choco-source.Tests.ps1:490]: @gep13 I'm going to resolve this conversation so it doesn't block merging, but tagging you for visibility.

**Review by corbob — APPROVED** [review]

**Comment by corbob** [comment]
Thanks for getting these added @gep13 and @vexx32 🎉

**Pull request merged**
